numpy/core/src/multiarray/ctors.c

```diff
@@ -2772,6 +2772,9 @@ PyArray_DescrFromObject(PyObject *op, PyArray_Descr *mintype)
 /* They all zero-out the memory as previously done */
 
 /* steals reference to descr -- and enforces native byteorder on it.*/
+
+/*Deprecated*/
```

Comment:
I think I would prefer this simply as part of the comment right above the function. Just replace the current comment (it is useless now anyway). (If you edit it, also please make sure it adheres to our comment style:
```
/*NUMPY_API
 * Comment comme
 * Which may be multiline
 */
```